app/controllers/course_summaries_controller.rb

```diff
@@ -2,17 +2,50 @@ class CourseSummariesController < ApplicationController
   include CourseSummariesHelper
 
   before_action :authorize_only_for_admin,
-                except: [:populate]
+                except: [:populate, :index]
 
   layout 'assignment_content'
 
   def index
   end
 
   def populate
+    averages = {}
+    medians = {}
+    visible_assessment_totals = []
+    if current_user.admin?
+      visible_assessments = Assessment.all.order(id: :asc).pluck_to_hash(:id, :type, :short_identifier)
```

Comment:
You don't need the `pluck_to_hash`. The rest of the query will return a relation that you can iterate over below (line 26). (So you won't need to call `find`)